jsettlers.common/src/jsettlers/common/position/ShortPoint2D.java

```diff
@@ -114,4 +114,11 @@ public static int getOnGridDist(final int dx, final int dy) {
 	public boolean equals(int x, int y) {
 		return this.x == x && this.y == y;
 	}
+
+	public double calculateDistanceTo(ShortPoint2D that) {
+		if (that == null) {
+			return 0;
+		}
+		return Math.sqrt((this.x - that.x) * (this.x - that.x) + (this.y - that.y) * (this.y - that.y));

```

Comment:
This is not the distance of two points, since we are working on a screwed grid.

Have a look at
jsettlers.common.map.shapes.MapCircle#getSquaredDistance(dx, dy)